### PR TITLE
COOK-26 disable dpkg autostart

### PR DESCRIPTION
**Pull request by wolf31o2**
- Introduced policy_rcd method to enable/disable auto-starting packages on debian-based distributions
- Update recipes to use a ruby_block to policy_rcd before/after package

### CODE REVIEW

File: spec/hive_metastore_spec.rb [hunk 1]
```diff
@@ -8,21 +8,31 @@
...
+      expect(chef_run).to run_ruby_block("package-#{pkg}")
+    end
+
+    it "creates #{pkg} service resource, but does not run it" do
+      expect(chef_run).to_not disable_service(pkg)
+      expect(chef_run).to_not enable_service(pkg)
+      expect(chef_run).to_not reload_service(pkg)
+      expect(chef_run).to_not restart_service(pkg)
+      expect(chef_run).to_not start_service(pkg)
+      expect(chef_run).to_not stop_service(pkg)
```
Comment (duddun) [inline spec/hive_metastore_spec.rb]: Why do we care if chef stops this service?

**Comment by duddun** [comment]
LGTM other than my comment above.

File: recipes/spark_historyserver.rb [hunk 2]
```diff
@@ -49,8 +63,17 @@
   action :nothing
 end
 
+if node['hadoop']['distribution'] == 'cdh'
+  s_cmd = "service #{pkg}"
+else
+  s_cmd = 'true #'
```
Comment (dereklwood) [inline recipes/spark_historyserver.rb]: extra ``#`` ?

Comment (dereklwood) [inline recipes/spark_historyserver.rb]: oh, nvm... i see... kinda wierd though

**Comment by dereklwood** [comment]
:+1:

File: spec/hive_metastore_spec.rb (hunk 1 above)
Comment (wolf31o2) [inline spec/hive_metastore_spec.rb]: We don't. We care that the code does what it is supposed to do. In this case, we want the code to do `:nothing` but there's no ChefSpec match for `:nothing` so we have to check for not $any_valid_action_other_than_nothing ... annoying, but "correct"

File: recipes/spark_historyserver.rb (hunk 2 above)
Comment (wolf31o2) [inline recipes/spark_historyserver.rb]: How about I add a comment to make it clearer?

**Pull request merged**